web/src/app/admin/actions/ActionTable.tsx

```diff
@@ -55,34 +58,50 @@ export function ActionsTable({
           {sortedMcpServers.map((server) => (
             <TableRow key={`mcp-${server.id}`}>
               <TableCell>
-                <div className="flex">
-                  <FiEdit2
-                    className="mr-1 my-auto cursor-pointer"
-                    onClick={() => {
-                      router.push(
-                        `/admin/actions/edit-mcp?server_id=${server.id}`
-                      );
-                    }}
-                  />
-                  <p className="text-text-04 font-medium whitespace-normal break-none">
+                <div className="flex items-center gap-2">
+                  {isAdmin || server.owner === user?.email ? (
+                    <button
+                      type="button"
+                      className="rounded p-1 hover:bg-background-neutral-02"
+                      onClick={() => {
+                        router.push(
+                          `/admin/actions/edit-mcp?server_id=${server.id}`
+                        );
+                      }}
+                    >
+                      <SvgEdit
+                        stroke="currentColor"
+                        className="size-4 text-text-03"
+                      />
+                    </button>
```

Comment:
Should definitely be using the `refresh-components/buttons/IconButton` here!

The `SvgEdit` can be placed in the `icon` prop passed to the `IconButton`:

```tsx
import IconButton from "@/refresh-components/buttons/IconButton";

<IconButton icon={SvgEdit} tertiary />
```

The `tertiary` flag is a prop that specifies the "variant" of the button. @duo-onyx has curated a very specific set of buttons and their states (hover states, active states, etc.) that we should try to adhere to as much as possible!